src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java

```diff
@@ -150,16 +150,17 @@ private void retryMessage(org.springframework.messaging.Message<OrderReceived> m
         String counterKey = receivedTopic + "-" + orderReceivedUri;
 
         if (receivedTopic.equals(ORDER_RECEIVED_TOPIC)
-                || retryCount.getOrDefault(counterKey, 0) >= MAX_RETRY_ATTEMPTS) {
+                || retryCount.getOrDefault(counterKey, 1) >= MAX_RETRY_ATTEMPTS) {
             republishMessageToTopic(orderReceivedUri, receivedTopic, nextTopic);
             if (!receivedTopic.equals(ORDER_RECEIVED_TOPIC)) {
                 resetRetryCount(counterKey);
             }
         }
         else {
-            retryCount.put(counterKey, retryCount.getOrDefault(counterKey, 0) + 1);
-            logMessageProcessingFailureRecoverable(message, retryCount.get(counterKey), ex);
+            retryCount.put(counterKey, retryCount.getOrDefault(counterKey, 1) + 1);
             // retry
+            handleMessage(message);
+            logMessageProcessingFailureRecoverable(message, retryCount.get(counterKey), ex);
```

Comment:
Why are we logging a failure here, after `handleMessage`?

Reply:
Good point. I've fixed this.